[css-images-4] Gradients with only one stop should be supported

roberto-apple opened the pull request:
#### 8c3641653fb098c034989b508663330a16e5d5b0
<pre>
[css-images-4] Gradients with only one stop should be supported
<a href="https://bugs.webkit.org/show_bug.cgi?id=285413">https://bugs.webkit.org/show_bug.cgi?id=285413</a>
<a href="https://rdar.apple.com/142796999">rdar://142796999</a>

Reviewed by Tim Nguyen.

Allow gradients to support only a single stop.
Update test expectations to reflect code change.
Add fuzziness to single stop tests to account for gradient noise.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/color-stops-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-004.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-005.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-006.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-007.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-008.html:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
(WebCore::CSSPropertyParserHelpers::consumeColorStopList):

Canonical link: <a href="https://commits.webkit.org/289469@main">https://commits.webkit.org/289469@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9df4b620fbb3b6d7f7a38d90569c399065a5797e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87024 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6534 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41379 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/91882 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/37763 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/89073 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/6810 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/14602 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67261 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25024 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90026 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5212 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/78770 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47583 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/4989 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/33145 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/36880 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/75481 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34022 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/93770 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14186 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/10323 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76066 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14390 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74623 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75265 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18530 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19604 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18031 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/7103 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14205 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/19498 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/13949 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17392 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/15730 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-001.html:
```diff
@@ -8,6 +8,7 @@
     <link rel="help" href="https://drafts.csswg.org/css-images-3/#color-stop-syntax">
     <link rel="help" href="https://drafts.csswg.org/css-images-3/#coloring-gradient-line">
     <meta name="assert" content="Tests that gradient with a single color stop renders the expected solid color">
+    <meta name="fuzzy" content="maxDifference=0-1; totalPixels=0-4000" />
```
nit: here and in the rest of the files, can you match the other meta tags?
```suggestion
    <meta name="fuzzy" content="maxDifference=0-1; totalPixels=0-4000">
```

nt1m reviewed (approved):
You might want to skip the 5 first tests on iOS in `LayoutTests/platform/ios/TestExpectations` or increase the fuzziness.

roberto-apple committed:
[css-images-4] Gradients with only one stop should be supported
https://bugs.webkit.org/show_bug.cgi?id=285413
rdar://142796999

Reviewed by Tim Nguyen.

Allow gradients to support only a single stop.
Update test expectations to reflect code change.
Add fuzziness to single stop tests to account for gradient noise.

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/color-stops-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-001.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-002.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-003.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-004.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-005.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-006.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-007.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-images/gradient/gradient-single-stop-008.html:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp:
(WebCore::CSSPropertyParserHelpers::consumeColorStopList):

Canonical link: https://commits.webkit.org/289469@main
